theano/compile/tests/test_nanguardmode.py

```diff
@@ -69,3 +69,14 @@ def test_NanGuardMode():
         assert_raises(AssertionError, fun, biga)  # big values
     finally:
         _logger.propagate = True
+
+    # Test with uint, we should not crash
+    x = T.matrix(dtype='uint8')
+    w = theano.shared(np.random.randn(5, 7).astype(theano.config.floatX))
+    y = T.dot(x, w)
+    a = (np.random.randn(3, 5).astype(theano.config.floatX) * 10).astype('uint8')
+    fun = theano.function(
+        [x], y,
+        mode=NanGuardMode(nan_is_error=True, inf_is_error=True)
+    )
```

Comment:
I just ran the test with the master of Theano, and even with `device=cuda` (which is not supposed to happen when running tests), it still passes.